internal/auth/tokenloader.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// CachedFileTokenLoader returns a function that loads a token from the given file path,
+// caches it for the specified interval, and reloads after the interval expires.
+// disable Reloading by setting interval to 0
+func cachedFileTokenLoader(path string, interval time.Duration, timeFn func() time.Time) func() (string, error) {
+	var (
+		mu          sync.Mutex
+		cachedToken string
+		lastRead    time.Time
+	)
+
+	return func() (string, error) {
+		mu.Lock()
+		defer mu.Unlock()
+
+		now := timeFn()
+
+		// Special case: interval = 0 means "never reload after first load"
+		if interval == 0 && !lastRead.IsZero() {
+			return cachedToken, nil
+		}
+
+		// Normal caching logic for non-zero intervals
+		if !lastRead.IsZero() && interval > 0 && now.Sub(lastRead) < interval {
+			return cachedToken, nil
+		}
```

Comment:
```suggestion
		// Special case: interval = 0 means "never reload after first load"
		// Otherwise reload only if `interval` time has passed since last load.
		if interval == 0 || (!lastRead.IsZero() && interval > 0 && now.Sub(lastRead) < interval) {
			return cachedToken, nil
		}
```